lib/SILOptimizer/LoopTransforms/ArrayBoundsCheckOpts.cpp

```diff
@@ -1097,6 +1097,38 @@ static bool isComparisonKnownTrue(BuiltinInst *Builtin, InductionInfo &IndVar) {
                                     m_Specific(IndVar.End)));
 }
 
+/// Based on the induction variable information this comparison is known to be
+/// false.
+static bool isComparisonKnownFalse(BuiltinInst *Builtin,
+                                   InductionInfo &IndVar) {
+  if (!IndVar.IsOverflowCheckInserted ||
+      IndVar.Cmp != BuiltinValueKind::ICMP_EQ)
+    return false;
+
+  // Eencountered a few false condition patterns we can detect and optimize:
```

Comment:
Also perhaps:

Pattern match a few false condition patterns that we  can detect and optimize: